src/FileIndexer.ts

```diff
@@ -97,42 +111,48 @@ export class FileIndexer {
     return symbol
   }
 
-  private visitSymbolOccurrence(node: ts.Node, sym: ts.Symbol): void {
-    const range = Range.fromNode(node).toLsif()
-    let role = 0
-    const isDefinition = this.declarationName(node.parent) === node
-    if (isDefinition) {
-      role |= scip.scip.SymbolRole.Definition
+  private visitDefinition(node: ts.Node, sym: ts.Symbol): void {
```

Comment:
I'm not convinced this refactoring is an improvement in maintainability. In the future, I want to add more symbol roles such as import and write and then the `symbol_roles` bitmask will need to be updated in two different places.